src/Variant.cpp

```diff
@@ -83,33 +83,347 @@
 
 #include "Variant.h"
 
+#include <memory>   // for std::unique_ptr()
+#include <sstream>
+#include <string>   // for std::stol()
+
+#include <cctype>   // for std::isdigit()
+
+namespace {
+
+template <typename T>
+std::ostream& vecToStream(std::ostream& out, std::vector<T> vec) {
+  out << "{";
+  bool first = true;
+  for (auto item : vec) {
+    if (!first) {
+      out << ", ";
+    }
+    first = false;
+    out << item;
+  }
+  out << "}";
+  return out;
+}
+
+std::ostream& stringToStream(std::ostream& out, std::string val) {
+  out << "string(len=" << val.size() << ", val=\"" << val << "\")";
+  return out;
+}
+
+std::ostream& vecStringToStream(std::ostream& out,
+                                std::vector<std::string> vec) {
+  out << "{";
+  bool first = true;
+  for (auto item : vec) {
+    if (!first) {
+      out << ", ";
+    }
+    first = false;
+    stringToStream(out, item);
+  }
+  out << "}";
+  return out;
+}
+
+// Tests if a[pos:pos+b.size()] == b[:]
+bool substrEquals(const std::string &a, size_t pos, const std::string &b) {
+  if (a.size() <= pos || a.size() - pos < b.size()) {
+    return false;
+  }
+  return std::equal(b.begin(), b.end(), a.begin() + pos);
+}
+
+template <typename T>
+std::vector<T> stringToVec(std::string str) {
```

Comment:
Describe what this does.  I can see it converts a comma-separated list of items into a vector of those items.  But it should be documented at least a little.